game/theater/iadsnetwork/iadsnetwork.py

```diff
@@ -249,17 +250,21 @@ def initialize_network_from_range(self) -> None:
                 except IadsNetworkException:
                     # TGO does not participate to iads network
                     continue
-                # Find nearby Power or Connection
-                for nearby_go in self.ground_objects.values():
-                    if nearby_go == go:
-                        continue
-                    if (
-                        IadsRole.for_category(go.category)
```

Comment:
This is the first mistake... `IadsRole.for_category(go.category)` will always return the same thing. To me it's pretty clear that `nearby_go` is the one we should be looking at instead.

Reply:
i do not really understand the comment, sorry

Reply:
Now i see what you mean after digging into the code. Would be great to have that a bit more explained in the comments. Also speaking about "mistakes" of others and pointing at them is not a really good habit just as side note - we all do mistakes and we do things for liberation as part of our hobby. So it is okay to expect to find things like this in a sandbox open source project. Especially in the `develop` branch.
Still, thanks for finding this one and fixing it! Good catch!

Reply:
> Now i see what you mean after digging into the code. Would be great to have that a bit more explained in the comments.

Yeah I was trying to clarify but you were faster :p
Next time I'll try to be a bit more precise in my explanation.

>Also speaking about "mistakes" of others and pointing at them is not a really good habit just as side note - we all do mistakes and we do things for liberation as part of our hobby.

I get what you're saying, but I don't quite agree with that mentality. As you point out, we all make mistakes, it's part of being human. However I think it's important to be made aware when a mistake was made, obviously in a polite manner. Reading back my comment I can see how it could be interpreted as condescending, but let me assure you that's not the case. In fact this is the type of mistake I think everyone makes from time to time, "brainfart" type of thing, which I find hilarious when that happens to me. Usually making me wonder what I was thinking at the time itself and reminding me of Einstein's quote about 2 things being infinite... 🤣